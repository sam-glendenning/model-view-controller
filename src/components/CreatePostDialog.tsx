import {
  Dialog,
  DialogTitle,
  DialogContent,
  Box,
  TextField,
  DialogActions,
  Button,
} from '@mui/material';
import React, { useCallback } from 'react';
import { useCreatePostController } from '@/controllers/useCreatePostController';

interface CreatePostDialogProps {
  isOpen: boolean;
  onClose: () => void;
  onSuccess?: (message: string) => void;
  onError?: (message: string) => void;
}

export const CreatePostDialog: React.FC<CreatePostDialogProps> = ({
  isOpen,
  onClose,
  onSuccess,
  onError,
}) => {
  const {
    formData,
    isSubmitDisabled,
    handleClose,
    updateTitle,
    updateBody,
    updateUserId,
    submitForm,
  } = useCreatePostController({
    onClose,
    onSuccess,
    onError,
  });

  const handleSubmit = useCallback(async () => {
    await submitForm();
  }, [submitForm]);

  return (
    <Dialog open={isOpen} onClose={handleClose} maxWidth="sm" fullWidth>
      <DialogTitle>Create New Post</DialogTitle>
      <DialogContent>
        <Box sx={{ pt: 1 }}>
          <TextField
            autoFocus
            margin="dense"
            label="Title"
            type="text"
            fullWidth
            variant="outlined"
            value={formData.title}
            onChange={e => updateTitle(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="Content"
            type="text"
            fullWidth
            multiline
            rows={4}
            variant="outlined"
            value={formData.body}
            onChange={e => updateBody(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="User ID"
            type="number"
            fullWidth
            variant="outlined"
            value={formData.userId}
            onChange={e => updateUserId(parseInt(e.target.value) ?? 1)}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={handleClose}>Cancel</Button>
        <Button
          onClick={handleSubmit}
          variant="contained"
          disabled={isSubmitDisabled}
        >
          Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};
