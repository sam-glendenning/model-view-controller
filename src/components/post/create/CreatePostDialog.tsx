import React from 'react';
import {
  Dialog,
  DialogTitle,
  DialogContent,
  Box,
  TextField,
  DialogActions,
  Button,
} from '@mui/material';
import type { Post } from '@/types';

interface CreatePostDialogProps {
  // Dialog state
  open: boolean;
  onClose: () => void;

  // Form data
  postData: Post;
  onTitleChange: (title: string) => void;
  onBodyChange: (body: string) => void;
  onUserIdChange: (userId: number) => void;

  // Actions
  onSubmit: () => Promise<void>;
  isCreating: boolean;
}

export const CreatePostDialog: React.FC<CreatePostDialogProps> = ({
  open,
  onClose,
  postData,
  onTitleChange,
  onBodyChange,
  onUserIdChange,
  onSubmit,
  isCreating,
}) => {
  return (
    <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
      <DialogTitle>Create New Post</DialogTitle>
      <DialogContent>
        <Box sx={{ pt: 1 }}>
          <TextField
            autoFocus
            margin="dense"
            label="Title"
            type="text"
            fullWidth
            variant="outlined"
            value={postData.title}
            onChange={e => onTitleChange(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="Content"
            type="text"
            fullWidth
            multiline
            rows={4}
            variant="outlined"
            value={postData.body}
            onChange={e => onBodyChange(e.target.value)}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="User ID"
            type="number"
            fullWidth
            variant="outlined"
            value={postData.userId}
            onChange={e => onUserIdChange(parseInt(e.target.value) ?? 1)}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>Cancel</Button>
        <Button onClick={onSubmit} variant="contained" disabled={isCreating}>
          Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};
