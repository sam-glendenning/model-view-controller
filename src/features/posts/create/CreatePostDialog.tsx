import React from 'react';
import {
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogTitle,
  TextField,
} from '@mui/material';
import type { Post } from '@/shared/types';

interface CreatePostDialogProps {
  // Dialog state
  open: boolean;
  onClose: () => void;

  // Form data
  postData: Omit<Post, 'id'>;
  onTitleChange: (title: string) => void;
  onBodyChange: (body: string) => void;
  onUserIdChange: (userId: number) => void;

  // Actions
  onSubmit: () => Promise<void>;
  isCreating: boolean;
}

export const CreatePostDialog: React.FC<CreatePostDialogProps> = ({
  open,
  onClose,
  postData,
  onTitleChange,
  onBodyChange,
  onUserIdChange,
  onSubmit,
  isCreating,
}) => {
  return (
    <Dialog open={open} onClose={onClose} maxWidth="sm" fullWidth>
      <DialogTitle>Create New Post</DialogTitle>
      <DialogContent>
        <Box sx={{ pt: 1 }}>
          <TextField
            autoFocus
            margin="dense"
            label="Title"
            type="text"
            fullWidth
            variant="outlined"
            value={postData.title}
            onChange={e => {
              onTitleChange(e.target.value);
            }}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="Content"
            type="text"
            fullWidth
            multiline
            rows={4}
            variant="outlined"
            value={postData.body}
            onChange={e => {
              onBodyChange(e.target.value);
            }}
            sx={{ mb: 2 }}
          />
          <TextField
            margin="dense"
            label="User ID"
            type="number"
            fullWidth
            variant="outlined"
            value={postData.userId}
            onChange={e => {
              const parsed = parseInt(e.target.value);
              onUserIdChange(Number.isNaN(parsed) ? 1 : parsed);
            }}
          />
        </Box>
      </DialogContent>
      <DialogActions>
        <Button onClick={onClose}>Cancel</Button>
        <Button
          onClick={() => void onSubmit()}
          variant="contained"
          disabled={isCreating}
        >
          Create
        </Button>
      </DialogActions>
    </Dialog>
  );
};
