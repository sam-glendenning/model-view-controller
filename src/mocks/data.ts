import type { Post, User } from '@/shared/types';

// Mock data
export const mockUsers: User[] = [
  {
    id: 1,
    name: 'John Doe',
    email: 'john.doe@example.com',
    phone: '555-0123',
    website: 'johndoe.com',
    company: {
      name: 'Doe Enterprises',
      catchPhrase: 'Innovation at its best',
      bs: 'synergistic solutions',
    },
    address: {
      street: '123 Main St',
      suite: 'Apt 1',
      city: 'Anytown',
      zipcode: '12345',
      geo: {
        lat: '40.7128',
        lng: '-74.0060',
      },
    },
  },
  {
    id: 2,
    name: 'Jane Smith',
    email: 'jane.smith@example.com',
    phone: '555-0456',
    website: 'janesmith.com',
    company: {
      name: 'Smith & Co',
      catchPhrase: 'Quality first',
      bs: 'scalable platforms',
    },
    address: {
      street: '456 Oak Ave',
      suite: 'Suite 200',
      city: 'Other City',
      zipcode: '67890',
      geo: {
        lat: '34.0522',
        lng: '-118.2437',
      },
    },
  },
];

export const mockPosts: Post[] = [
  {
    id: '1',
    userId: 1,
    title: 'Test Post Title',
    body: 'This is a test post body with some content to display.',
  },
  {
    id: '2',
    userId: 1,
    title: 'Another Test Post',
    body: 'This is another test post with different content.',
  },
  {
    id: '3',
    userId: 2,
    title: "Jane's Post",
    body: 'A post written by Jane Smith about various topics.',
  },
  {
    id: '4',
    userId: 1,
    title: 'Understanding React Hooks',
    body: 'React Hooks have revolutionized how we write components. This post explores the most commonly used hooks and their practical applications.',
  },
  {
    id: '5',
    userId: 2,
    title: 'Building Scalable APIs',
    body: 'A comprehensive guide to building APIs that can handle growth and maintain performance as your application scales.',
  },
];
