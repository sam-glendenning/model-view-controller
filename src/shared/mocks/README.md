# MSW Mock Setup Architecture

This project uses different MSW handler strategies for development vs testing to address different requirements.

## Handler Types

### 1. Runtime Handlers (`handlers.ts`)

**Purpose**: Development experience with persistent data

- **Stateful**: Mutations persist during development session
- **Realistic**: Full CRUD operations modify shared state
- **Session continuity**: Data changes visible across page refreshes

**Usage**:

- Development server (`yarn dev`)
- Manual testing and UI development

### 2. Test Handlers (`testHandlers.ts`)

**Purpose**: Isolated, deterministic testing

- **Stateless**: No mutations to global mock data
- **Isolated**: Each test gets fresh data copies
- **Predictable**: Tests don't affect each other

**Usage**:

- Automated tests (`yarn test`)
- CI/CD pipelines

## Why This Separation?

### The Problem

Originally, both development and tests used the same handlers, which caused:

- **Test pollution**: One test's mutations affected subsequent tests
- **Flaky tests**: Tests passed/failed depending on execution order
- **Hard to debug**: Mysterious state changes between tests

### The Solution

- **Development**: Keep stateful handlers for realistic UX
- **Testing**: Use stateless handlers for test isolation
- **Clean separation**: Different needs, different implementations

## Best Practices

### For Real Applications

In production apps:

- **Development**: Use MSW for consistent development experience
- **Testing**: Use MSW with isolated handlers (like this project)
- **Production**: Real API calls (no MSW)

### Handler Guidelines

- **Runtime handlers**: Mutate shared state for persistence
- **Test handlers**: Return isolated copies, never mutate globals
- **Always**: Deep clone mock data in test handlers
