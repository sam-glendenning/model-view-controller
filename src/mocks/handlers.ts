import { http, HttpResponse, delay } from 'msw';
import type { Post } from '@/shared/types';
import { mockUsers, mockPosts } from './data';
import { v4 as uuidv4 } from 'uuid';

// Artificial delay function
const artificialDelay = () => delay(1000);

// Request handlers with artificial delays
export const handlers = [
  // Users
  http.get('https://jsonplaceholder.typicode.com/users', async () => {
    await artificialDelay();
    return HttpResponse.json(mockUsers);
  }),

  http.get(
    'https://jsonplaceholder.typicode.com/users/:id',
    async ({ params }) => {
      await artificialDelay();
      const id = parseInt(params['id'] as string);
      const user = mockUsers.find(u => u.id === id);

      if (!user) {
        return new HttpResponse(null, { status: 404 });
      }

      return HttpResponse.json(user);
    },
  ),

  // Posts
  http.get(
    'https://jsonplaceholder.typicode.com/posts',
    async ({ request }) => {
      await artificialDelay();
      const url = new URL(request.url);
      const userId = url.searchParams.get('userId');

      if (userId) {
        const filteredPosts = mockPosts.filter(
          p => p.userId === parseInt(userId),
        );
        return HttpResponse.json(filteredPosts);
      }

      return HttpResponse.json(mockPosts);
    },
  ),

  http.get(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params }) => {
      await artificialDelay();
      const id = params['id'] as string;
      const post = mockPosts.find(p => p.id === id);

      if (!post) {
        return new HttpResponse(null, { status: 404 });
      }

      return HttpResponse.json(post);
    },
  ),

  http.post(
    'https://jsonplaceholder.typicode.com/posts',
    async ({ request }) => {
      await artificialDelay();
      const newPost = (await request.json()) as Omit<Post, 'id'>;
      // Generate a unique UUID for the new post
      const post: Post = {
        ...newPost,
        id: uuidv4(),
      };

      mockPosts.push(post);
      return HttpResponse.json(post, { status: 201 });
    },
  ),

  http.put(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params, request }) => {
      await artificialDelay();
      const id = params['id'] as string;
      const updatedData = (await request.json()) as Partial<Post>;
      const postIndex = mockPosts.findIndex(p => p.id === id);

      if (postIndex === -1) {
        return new HttpResponse(null, { status: 404 });
      }

      const existingPost = mockPosts[postIndex]!;

      const updatedPost: Post = {
        id: existingPost.id,
        userId: updatedData.userId ?? existingPost.userId,
        title: updatedData.title ?? existingPost.title,
        body: updatedData.body ?? existingPost.body,
      };

      mockPosts[postIndex] = updatedPost;
      return HttpResponse.json(updatedPost);
    },
  ),

  http.delete(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params }) => {
      await artificialDelay();
      const id = params['id'] as string;
      const postIndex = mockPosts.findIndex(p => p.id === id);

      if (postIndex === -1) {
        return new HttpResponse(null, { status: 404 });
      }

      mockPosts.splice(postIndex, 1);
      return new HttpResponse(null, { status: 200 });
    },
  ),
];
