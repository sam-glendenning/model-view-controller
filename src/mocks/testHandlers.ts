import { http, HttpResponse } from 'msw';
import type { Post, User } from '@/shared/types';
import { mockUsers, mockPosts } from './data';
import { v4 as uuidv4 } from 'uuid';

// Create isolated copies for each test handler call to prevent mutations
const getIsolatedMockPosts = (): Post[] =>
  JSON.parse(JSON.stringify(mockPosts));
const getIsolatedMockUsers = (): User[] =>
  JSON.parse(JSON.stringify(mockUsers));

// Test handlers with no delay and NO shared state mutations
export const testHandlers = [
  // Users
  http.get('https://jsonplaceholder.typicode.com/users', async () => {
    return HttpResponse.json(getIsolatedMockUsers());
  }),

  http.get(
    'https://jsonplaceholder.typicode.com/users/:id',
    async ({ params }) => {
      const id = parseInt(params['id'] as string);
      const users = getIsolatedMockUsers();
      const user = users.find(u => u.id === id);

      if (!user) {
        return new HttpResponse(null, { status: 404 });
      }

      return HttpResponse.json(user);
    },
  ),

  // Posts
  http.get(
    'https://jsonplaceholder.typicode.com/posts',
    async ({ request }) => {
      const url = new URL(request.url);
      const userId = url.searchParams.get('userId');
      const posts = getIsolatedMockPosts();

      if (userId) {
        const filteredPosts = posts.filter(p => p.userId === parseInt(userId));
        return HttpResponse.json(filteredPosts);
      }

      return HttpResponse.json(posts);
    },
  ),

  http.get(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params }) => {
      const id = params['id'] as string;
      const posts = getIsolatedMockPosts();
      const post = posts.find(p => p.id === id);

      if (!post) {
        return new HttpResponse(null, { status: 404 });
      }

      return HttpResponse.json(post);
    },
  ),

  http.post(
    'https://jsonplaceholder.typicode.com/posts',
    async ({ request }) => {
      const newPost = (await request.json()) as Omit<Post, 'id'>;
      // Generate a unique UUID for the new post
      const post: Post = {
        ...newPost,
        id: uuidv4(),
      };

      // Return the new post WITHOUT mutating global state
      return HttpResponse.json(post, { status: 201 });
    },
  ),

  http.put(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params, request }) => {
      const id = params['id'] as string;
      const updatedData = (await request.json()) as Partial<Post>;
      const posts = getIsolatedMockPosts();
      const postIndex = posts.findIndex(p => p.id === id);

      if (postIndex === -1) {
        return new HttpResponse(null, { status: 404 });
      }

      // Create updated post WITHOUT mutating global state
      const updatedPost = { ...posts[postIndex], ...updatedData };
      return HttpResponse.json(updatedPost);
    },
  ),

  http.delete(
    'https://jsonplaceholder.typicode.com/posts/:id',
    async ({ params }) => {
      const id = params['id'] as string;
      const posts = getIsolatedMockPosts();
      const postIndex = posts.findIndex(p => p.id === id);

      if (postIndex === -1) {
        return new HttpResponse(null, { status: 404 });
      }

      // Return success WITHOUT mutating global state
      return new HttpResponse(null, { status: 200 });
    },
  ),
];
