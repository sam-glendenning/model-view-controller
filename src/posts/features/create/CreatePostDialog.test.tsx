import { render } from '@testing-library/react';
import { CreatePostDialog } from './CreatePostDialog';
import type { Post } from '@/posts/types';
import { mockPosts } from '@/shared/mocks/data';

describe('CreatePostDialog', () => {
  let mockPost: Omit<Post, 'id'>;

  beforeEach(() => {
    jest.clearAllMocks();

    mockPost = {
      title: mockPosts[0]!.title,
      body: mockPosts[0]!.body,
      userId: mockPosts[0]!.userId,
    };
  });

  it('renders correctly', () => {
    const { baseElement } = render(
      <CreatePostDialog
        open={true}
        postData={mockPost}
        onTitleChange={jest.fn()}
        onBodyChange={jest.fn()}
        onUserIdChange={jest.fn()}
        onSubmit={jest.fn()}
        onClose={jest.fn()}
        isCreating={false}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });
});
