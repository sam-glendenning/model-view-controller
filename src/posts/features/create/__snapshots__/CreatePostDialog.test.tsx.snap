// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`CreatePostDialog renders correctly 1`] = `
<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="MuiDialog-root MuiModal-root css-1424xw8-MuiModal-root-MuiDialog-root"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiModal-backdrop css-1cennmq-MuiBackdrop-root-MuiDialog-backdrop"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiDialog-container MuiDialog-scrollPaper css-19do60a-MuiDialog-container"
      role="presentation"
      style="opacity: 1; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      tabindex="-1"
    >
      <div
        aria-labelledby="«r0»"
        aria-modal="true"
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation24 MuiDialog-paper MuiDialog-paperScrollPaper MuiDialog-paperWidthSm MuiDialog-paperFullWidth css-16htz67-MuiPaper-root-MuiDialog-paper"
        role="dialog"
        style="--Paper-shadow: 0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12);"
      >
        <h2
          class="MuiTypography-root MuiTypography-h6 MuiDialogTitle-root css-rvoa5x-MuiTypography-root-MuiDialogTitle-root"
          id="«r0»"
        >
          Create New Post
        </h2>
        <div
          class="MuiDialogContent-root css-kw13he-MuiDialogContent-root"
        >
          <div
            class="MuiBox-root css-zoser8"
          >
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1fksspf-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="«r1»"
                id="«r1»-label"
              >
                Title
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth Mui-focused MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  id="«r1»"
                  type="text"
                  value="Test Post Title"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ex8a5f-MuiNotchedOutlined-root"
                  >
                    <span>
                      Title
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-1fksspf-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="«r2»"
                id="«r2»-label"
              >
                Content
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline css-zsp0bh-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <textarea
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                  id="«r2»"
                  rows="4"
                  style="height: 0px; overflow: hidden;"
                >
                  This is a test post body with some content to display.
                </textarea>
                <textarea
                  aria-hidden="true"
                  class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline css-w4nesw-MuiInputBase-input-MuiOutlinedInput-input"
                  readonly=""
                  style="visibility: hidden; position: absolute; overflow: hidden; height: 0px; top: 0px; left: 0px; transform: translateZ(0); padding-top: 0px; padding-bottom: 0px; width: 100%;"
                  tabindex="-1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ex8a5f-MuiNotchedOutlined-root"
                  >
                    <span>
                      Content
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth MuiTextField-root css-xanwcj-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined css-113d811-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="«r3»"
                id="«r3»-label"
              >
                User ID
              </label>
              <div
                class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-1blp12k-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiInputBase-input MuiOutlinedInput-input css-16wblaj-MuiInputBase-input-MuiOutlinedInput-input"
                  id="«r3»"
                  type="number"
                  value="1"
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-18p5xg2-MuiNotchedOutlined-root-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-ex8a5f-MuiNotchedOutlined-root"
                  >
                    <span>
                      User ID
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiDialogActions-root MuiDialogActions-spacing css-15fu35s-MuiDialogActions-root"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-colorPrimary css-1uent87-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Cancel
          </button>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-74d805-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Create
          </button>
        </div>
      </div>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
